app/controllers/sign_up/recovery_codes_controller.rb

```diff
@@ -6,7 +6,11 @@ class RecoveryCodesController < ApplicationController
     before_action :confirm_has_not_already_viewed_recovery_code
 
     def show
-      user_session.delete(:first_time_recovery_code_view)
+      if user_session[:first_time_recovery_code_view].present?
+        @show_progress_bar = true
+        user_session.delete(:first_time_recovery_code_view)
```

Comment:
is `user_session` a hash or hash-like? if so, the `.delete` will return the value so you could do:

```ruby
if user_session.delete(:first_time_recovery_code_view).present?
  @show_progress_bar = true
end
```

but rubocop might get mad so it might be:


```ruby
@show_progress_bar = true if user_session.delete(:first_time_recovery_code_view).present?
```